sdk/core/Azure.Core/src/Shared/ConnectionString.cs

```diff
@@ -34,6 +38,12 @@ private ConnectionString(Dictionary<string, string> pairs, string pairSeparator,
         public string? GetNonRequired(string keyword) =>
             _pairs.TryGetValue(keyword, out var value) ? value : null;
 
+        public bool TryGetSegmentValue(string keyword, out string value) =>
```

Comment:
It may be a good idea to add a few tests for new methods and the fix https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/tests/ConnectionStringTests.cs

Reply:
Thanks - I tend to forget that tests don't show up as referenced callers for shared source.